com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UploadAppStep.java

```diff
@@ -67,16 +69,27 @@ public StepPhase executeAsyncStep(ExecutionWrapper execution) throws FileStorage
             return StepPhase.DONE;
         }
 
-        getStepLogger().debug(Messages.UPLOADING_FILE_0_FOR_APP_1, fileName, app.getName());
-
         String newApplicationDigest = getNewApplicationDigest(execution, appArchiveId, fileName);
-        detectApplicationFileDigestChanges(execution, app.getName(), client, newApplicationDigest);
+        boolean contentChanged = detectApplicationFileDigestChanges(execution, appName, client, newApplicationDigest);
+        if (!contentChanged && isAppStagedCorrectly(client, appName)) {
+            getStepLogger().info(Messages.CONTENT_OF_APPLICATION_0_IS_NOT_CHANGED, appName);

```

Comment:
Is this necessarry to be info message?

Reply:
I think it is much more easier to understand what is happening with your deployment when this message exists as otherwise you only see the message saying "Uploading Application X", while there is no actual deployment happening.